plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -129,13 +129,8 @@ public static HiveMetastore cachingHiveMetastore(HiveMetastore delegate, Executo
                 config.getMetastoreCacheMaximumSize());
     }
 
-    public static HiveMetastore cachingHiveMetastore(HiveMetastore delegate, Executor executor, Duration cacheTtl, Optional<Duration> refreshInterval, long maximumSize)
+    public static CachingHiveMetastore cachingHiveMetastore(HiveMetastore delegate, Executor executor, Duration cacheTtl, Optional<Duration> refreshInterval, long maximumSize)
     {
-        if (cacheTtl.toMillis() == 0 || maximumSize == 0) {
```

Comment:
check state that caching is enabled

Reply:
This code is now moved to `CachingHiveMetastoreModule` and is implemented as checking if enabled.

Reply:
Still, these methods are used outside of `CachingHiveMetastoreModule` (e.g. in tests), so extra check makes sense

Reply:
Hmm Earlier methods where returning HiveMetastore in such cases. 
Now I can only throw exception here WDYT ?

Reply:
Added checkState